### PR TITLE
Fix query for filtering by failed jobs

### PR DESCRIPTION
**Pull request by hmac**
Also add a load of specs

Depends on https://github.com/gocardless/que_ui/pull/3

### CODE REVIEW

File: lib/que_ui/queries.rb [hunk 1]
```diff
@@ -23,30 +25,31 @@ module Queries
       priority, lower(job_class)
     SQL
 
-    LOCK_ID = "((pg_locks.classid::bigint << 32) | pg_locks.objid::bigint)"
     INACTIVE = "(s.pg_lock_id IS NULL AND NOT s.retryable)"
-    PENDING_RETRY = "(s.pg_lock_id IS NOT NULL AND s.error_count > 0 AND s.retryable)"
+    PENDING_RETRY = "(s.pg_lock_id IS NULL AND s.error_count > 0 AND s.retryable)"
```
Comment (Sinjo) [inline lib/que_ui/queries.rb]: Dangit!

File: spec/support/job.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class FailingJob < Que::Job
+  include Que::Failure::NoRetry
+
+  def run(*args)
+    raise "Oh noes!"
+  end
+end
+
+class RetryableJob < Que::Job
```
Comment (Sinjo) [inline spec/support/job.rb]: I'd include `VariableRetry` here as we don't use Que's default retry behaviour.

Comment (hmac) [inline spec/support/job.rb]: Yeah I think that makes sense, although there's nothing stopping someone writing a job that doesn't use `NoRetry` or `VariableRetry` and expecting QueUI to work with it... will leave that problem for another day though.

Comment (Sinjo) [inline spec/support/job.rb]: In our codebase there is (Rubocop), but in general (if someone else were using this) there isn't.

**Pull request merged**